src/analysis/souffle/taint.dl

```diff
@@ -99,12 +99,18 @@ mayHaveTag(tgt, owner, tag) :-
 isIntegrityTag(integTag) :- hasAppliedIntegrityTag(_, _, integTag).
 isPrincipal(principal) :- hasAppliedIntegrityTag(_, principal, _).
 
+.decl mult_op(result: AccessPath, operand1: AccessPath, operand2: AccessPath)
+.decl div_op(result: AccessPath, dividend: AccessPath, divisor: AccessPath)
+
 // Any access point that has no incoming edges and is not directly given an
 // integrity tag lacks that integrity tag, as there is no way for it to
 // propagate to that AccessPath.
 lacksIntegrityTag(ap, owner, integTag) :-
   isAccessPath(ap), isPrincipal(owner), isIntegrityTag(integTag),
-  !edge(_, ap), !hasAppliedIntegrityTag(ap, owner, integTag).
+  // We will need to capture the fact that some propagation happens because of
+  // operators.
+  !edge(_, ap), !mult_op(ap, _, _), !div_op(ap, _, _),
```

Comment:
In a real system, we'd probably make all of these ops put entries in an `operator` relation and check that one relation rather than each of these specifically. But I get that this is an example PR for the sake of discussion, just flagging this issue in case it becomes the basis for a real PR.